pkgs/by-name/fi/finamp/package.nix

```diff
@@ -0,0 +1,71 @@
+{ lib
+, flutter
+, mpv
+, xdg-user-dirs
+, patchelf
+, fetchFromGitHub
+, makeDesktopItem
+}:
+let
+  version = "0.9.8-beta";
+in
+flutter.buildFlutterApplication {
+  inherit version;
+  pname = "finamp";
+  src = fetchFromGitHub {
+    owner = "jmshrv";
+    repo = "finamp";
+    rev = version;
+    hash = "sha256-lvjhA+hdCXgDsrNhNw4Tiq6ZgkYlPuMeHha8OJNF1TI=";
+  };
+  pubspecLock = lib.importJSON ./pubspec.lock.json;
+
+  nativeBuildInputs = [ patchelf ];
+  buildInputs = [ mpv xdg-user-dirs ];
+
+  gitHashes = {
+    balanced_text = "sha256-lSDR5dDjZ4garRbBPI+wSxC5iScg8wVSD5kymmLbYbk=";
+    isar_generator = "sha256-lWnHmZmYx7qDG6mzyDqYt+Xude2xVOH1VW+BoDCas60=";
+    media_kit_libs_windows_audio = "sha256-p3hRq79whLFJLNUgL9atXyTGvOIqCbTRKVk1ie0Euqs=";
+    palette_generator = "sha256-mnRJf3asu1mm9HYU8U0di+qRk3SpNFwN3S5QxChpIA0=";
+    split_view = "sha256-unTJQDXUUPVDudlk0ReOPNYrsyEpbd/UMg1tHZsmg+k=";
+  };
+
+  postFixup = ''
+    patchelf $out/app/finamp --add-needed libisar.so --add-needed libmpv.so --add-rpath ${lib.makeLibraryPath [ mpv ]}
+  '';
+
+  postInstall = ''
+    d=$out/share/icons/hicolor/scalable/apps
+    mkdir -p $d
+    cp $src/assets/icon/icon_foreground.svg $d/finamp.svg
+
+    mkdir -p $out/share/applications
+    cp $desktopItem/share/applications/*.desktop $out/share/applications
+  '';
+
+  desktopItem = makeDesktopItem {
+    name = "Finamp";
+    desktopName = "Finamp";
+    genericName = "Music Player";
+    exec = "finamp";
+    icon = "finamp";
+    startupWMClass = "finamp";
+    comment = "An open source Jellyfin music player";
+    categories = [
+      "AudioVideo"
+      "Audio"
+      "Player"
+      "Music"
+    ];
+  };
+
+  meta = with lib; {
```

Comment:
Better to avoid `with lib;`

Reply:
That’s not the first time I see another reviewer say this. What is the rationale? At such a low level, it doesn’t introduce much hassle to understand the code. Would it be placed at a higher level I would agree with you, but I don’t understand why it would be an issue here.

Reply:
See https://github.com/NixOS/nixpkgs/issues/208242

Reply:
Thanks for pointing to this issue. As I said earlier, I do understand and do agree that using `with` at a high level is clearly misleading and makes the code much harder to read, even more when multiple `with` are used.

However, here I can only agree with SuperSandro and eclairevoyant:

> The problem is not when with is used in meta where it is very local and only covering a few lines. I am completely against changing those treewide.

> The nix.dev antipatterns guide warns against using with at the top of a Nix file - a reasonable suggestion, but does not mention anything about using with in the very predictable/standardised spot of only in meta.

I believe the `with` usages on `meta` attributes are legit since they save a few keystrokes while keeping the code readable, even for beginners. IMO this should not be changed.

Reply:
I will change it to match the manual at https://nixos.org/manual/nixpkgs/unstable/#chap-meta